docs/tutorial.md

```diff
@@ -0,0 +1,276 @@
+# Tutorial
+
+We'll create a super simple offline-first, remotely synchronized todo list application, step-by-step. End result will look like this:
+
+- XXX: final app screenshot
+
+## Prerequisites
+
+To remotely store and synchronize data, we're using a [Kinto](http://kinto.readthedocs.org/) server instance on the [0.2.2 tag](https://github.com/mozilla-services/kinto/releases/tag/0.2.2). Please refer to Kinto documentation for [setup instructions](http://kinto.readthedocs.org/en/stable/installation.html).
+
+## Bootstrapping a demo Web application
+
+First, let's create a simple HTML file for our demo app, in an `index.html` file:
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Cliquetis demo</title>
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
+</head>
+<body>
+  <div class="container">
+    <h1>Tasks</h1>
+    <form class="form-inline" id="form">
+      <div class="form-group">
+        <input class="form-control" type="text" name="label" placeholder="Thing">
+        <input class="btn btn-primary" type="submit" value="Add">
+      </div>
+    </form>
+    <ul id="tasks" class="list-group"></ul>
+  </div>
+  <script src="https://raw.githubusercontent.com/mozilla-services/cliquetis/master/dist/cliquetis.min.js"></script>
+  <script src="demo.js"></script>
+</body>
+</html>
+```
+
+For now, our `demo.js` file content is simply:
+
+```js
+function main() {
+  var db = new Cliquetis();
+  var tasks = db.collection("tasks");
+}
+
+window.addEventListener("DOMContentLoaded", main);
+
+```
+
+**Note:** You'll need to serve this page over HTTP, for Cliquetis to work. To do so, you can use node's [http-server](https://github.com/indexzero/http-server), Python's [SimpleHTTPServer](https://docs.python.org/2/library/simplehttpserver.html) or whatever Web server you like.
+
+For example, if you're using http-server:
+
+    $ npm install -g http-server
+    $ http-server yourdirectory
+
+And that's it. You should see something like this:
+
+![](images/step1.png)
+
+## Adding new tasks
+
+We want to listen to form submission events to add tasks into our local database. In `demo.js`, we write:
+
+```js
+function main() {
+  var db = new Cliquetis();
+  var tasks = db.collection("tasks");
+
+  document.getElementById("form")
+    .addEventListener("submit", function(event) {
+      event.preventDefault();
+      tasks.create({
+        label: event.target.label.value,
+        done: false
+      }).then(res => {
```

Comment:
Isn't better indented like this ?
(code layout is like the hardest part of Promises :))

```
    tasks.create({
        label: event.target.label.value,
        done: false
      })
      .then(res => {
```